File: leetcode/14. Longest Common Prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if(strs.size() == 0){
            return "";
        }
        
        string pre = "";
        
        for(int i = 0; i < strs[0].size(); ++i){
            for(auto str : strs){
                if(str.size() - 1 < i){
                    return pre;
                }
                if(str[i] != strs[0][i]){
                    return pre;
                }
            }
            pre += strs[0][i];
        }
        return pre;
        
    }
};

File: leetcode/14. Longest Common Prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        idx = 0
        for i in range(len(strs[0])):
            c = strs[0][i]
            advance = True
            for word in strs:
                if len(word)-1 < i or word[i] != c:
                    advance = False
                    break
            if not advance:
                break
            idx += 1
        return strs[0][:idx]
